### ISSUE TITLE
Validate 2.1.0-SNAPSHOT skips a collection XML label

### ISSUE THREAD

**Issue by jennifergward** [NASA-PDS-Incubator/validate#408]
## 🐛 Describe the bug
The Validate tool is reporting that it's skipping collection_data_table.xml "due to collection not latest or does not sharing the same logical_identifier as the bundle target." The LID is the same in the collection product and the bundle, so the problem must be “collection not latest”. Since the collection product is listed in the bundle by its LID, not LIDVID, the rule is to assume the latest version of the collection is intended. So what makes Validate think this version is not the latest? To complicate matters, a few lines down the Validate tool reports that collection_data_table.xml passed. 

## 📜 To Reproduce
The bundle in question is the InSight SEIS bundle at https://pds-geosciences.wustl.edu/insight/urn-nasa-pds-insight_seis/data/. The bundle is too large to send a copy, so I'm attaching the Validate report, bundle.xml, and collection product.

## 🕵️ Expected behavior
I expect Validate not to skip the collection.

## 📚 Version of Software Used
Validate 2.1.0-SNAPSHOT

## 🩺 Test Data / Additional context
[validate.zip](https://github.com/NASA-PDS/validate/files/7226301/validate.zip)

##  🏞Screenshots
n/a

## 🖥 System Info
OS: Windows 10

---

**Comment by qchaupds** [comment]
@jennifergward @jordanpadams Validate looks for the logical_identifier plus the version_id in the collection label for latest.  Is it possible that the same logical_identifier "urn:nasa:pds:insight_seis:data_table" is used in another collection file?

Is there a way for you to "grep" for the value "urn:nasa:pds:insight_seis:data_table" in all the collection labels?

In DEV, we have two 'data_seed' collections that shares the same logical_identifier but different version.  Validate correctly ignore the version that is not the latest:

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 119 % validate -R pds4.bundle -r report_github408_bundle_valid.json -s json -t src/test/resources/github408/valid/bundle_insight_seis.xml 

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 112 % grep -i skip  report_github408_bundle_valid.json
      "status": "SKIP",
          "message": "Skipping file:/home/qchau/sandbox/validate/src/test/resources/github408/valid/data/collection_data_seed_2.xml due to collection not latest or does not sharing the same logical_identifier as the bundle target"
```

These commands show we have two identical collection with different version number.

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 117 % ls -l /home/qchau/sandbox/validate/src/test/resources/github408/valid/data | grep collection | grep data_seed
-rw-r--r-- 1 qchau pds    8044 Sep 27 10:41 collection_data_seed_2.xml
-rw-r--r-- 1 qchau pds    8045 Sep 27 10:40 collection_data_seed.xml

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 118 % grep -A 1 logical /home/qchau/sandbox/validate/src/test/resources/github408/valid/data/collection_data_seed*.xml

/home/qchau/sandbox/validate/src/test/resources/github408/valid/data/collection_data_seed_2.xml:        <logical_identifier>urn:nasa:pds:insight_seis:data_seed</logical_identifier>
/home/qchau/sandbox/validate/src/test/resources/github408/valid/data/collection_data_seed_2.xml-        <version_id>9.0</version_id>
--
/home/qchau/sandbox/validate/src/test/resources/github408/valid/data/collection_data_seed.xml:        <logical_identifier>urn:nasa:pds:insight_seis:data_seed</logical_identifier>
/home/qchau/sandbox/validate/src/test/resources/github408/valid/data/collection_data_seed.xml-        <version_id>10.0</version_id>
```

The version with value 9.0 in collection_data_seed_2.xml was correctly skipped.

The 2nd issue you had raised about the collection still being validated is another bug being looked at.

**Comment by jordanpadams** [comment]
@qchaupds from the URL provided, it looks like it is unlikely that the LID is duplicated (there are only 3 collections in the directory provided). Could the issue have something to do with the collections being in the same directory? or can we try downloading that `data_table` file and see if there is some funky windows thing happening where we are unable to read the file correctly?

**Comment by qchaupds** [comment]
@jordanpadams The issue is more complicated now that there is another bug at the same time.  Stay tuned.  It will take a while to unravel this web.

**Comment by qchaupds** [comment]
@jordanpadams Just discovered the bug.  It has to do with finding the largest version number.  Fixing it is another issue.  More to come in the coming days.

**Comment by tloubrieu-jpl** [comment]
Not merged yet
